oidc-api/src/main/java/uk/gov/di/authentication/oidc/lambda/AuthCodeHandler.java

```diff
@@ -365,19 +370,28 @@ private AuthorizationCode generateAuthCode(
             URI redirectUri,
             ClientSession clientSession,
             String clientSessionId,
-            Session session)
+            Session session,
+            OrchSessionItem orchSession)
             throws ClientNotFoundException, ProcessAuthRequestException {
         if (!orchestrationAuthorizationService.isClientRedirectUriValid(clientID, redirectUri)) {
             throw new ProcessAuthRequestException(400, ErrorResponse.ERROR_1016);
         }
         CredentialTrustLevel lowestRequestedCredentialTrustLevel =
                 VectorOfTrust.getLowestCredentialTrustLevel(clientSession.getVtrList());
-        if (isNull(session.getCurrentCredentialStrength())
-                || lowestRequestedCredentialTrustLevel.compareTo(
-                                session.getCurrentCredentialStrength())
-                        > 0) {
+        var currentCredentialStrength = session.getCurrentCredentialStrength();
+        if (isNull(currentCredentialStrength)
+                || lowestRequestedCredentialTrustLevel.compareTo(currentCredentialStrength) > 0) {
             session.setCurrentCredentialStrength(lowestRequestedCredentialTrustLevel);
         }
+
+        if (configurationService.isCurrentCredentialStrengthInOrchSessionEnabled()
+                && (isNull(currentCredentialStrength)
+                        || lowestRequestedCredentialTrustLevel.compareTo(currentCredentialStrength)
+                                > 0)) {
+            orchSessionService.updateSession(
+                    orchSession.withCurrentCredentialStrength(lowestRequestedCredentialTrustLevel));
+        }
```

Comment:
Would it be neater to nest these check so we don't need to make the same comparison twice?

Reply:
Deliberately kept this separate as in the next PR we will be using the orchSession to the do comparison.